frontend/src/utils/blockUtils.js

```diff
@@ -33,6 +33,13 @@ export function generateId(block) {
   return id;
 }
 
+export function generatePipelineId(pipeline) {
+  const nanoid = customAlphabet("1234567890abcdefghijklmnopqrstuvwxyz", 12);
+  const newNanoid = nanoid();
+  const id = `${pipeline.name}-${newNanoid}`;
+  return id;
+}
```

Comment:
This seems very similar to the `generateId` function in that file do you think there's any way to reuse the code?